contrib/timelapse.lua

```diff
@@ -0,0 +1,463 @@
+local gettext = dt.gettext
+
+dt.configuration.check_version(...,{5,0,0})
+
+local MODULE_NAME = 'timelapse'
+gettext.bindtextdomain(MODULE_NAME, dt.configuration.config_dir..'/lua/')
+
+local function _(msgid)
+  return gettext.dgettext(MODULE_NAME, msgid)
+end
+
+local OS_PATH_SEPARATOR = dt.configuration.running_os == "windows" and  "\\"  or  "/"
+
+---- DECLARATIONS
+
+local resolutions = {
+  ['QVGA'] = {
+    ['label'] = 'QVGA 320x240 (4:3)',
+    ['w'] = 320,
+    ['h'] = 240
+  },
+  ['HVGA'] = {
+    ['label'] = 'HVGA 480x320 (3:2)',
+    ['w'] = 480,
+    ['h'] = 320
+  },
+  ['VGA'] = {
+    ['label'] = 'VGA 640x480 (4:3)',
+    ['w'] = 640,
+    ['h'] = 480
+  },
+  ['HDTV 720p'] = {
+    ['label'] = 'HDTV 720p 1280x720 (16:9)',
+    ['w'] = 1280,
+    ['h'] = 720
+  },
+  ['HDTV 1080p'] = {
+    ['label'] = 'HDTV 1080p 1920x1080 (16:9)',
+    ['w'] = 1920,
+    ['h'] = 1080
+  },
+  ['Cinema TV'] = {
+    ['label'] = 'Cinema TV 2560x1080 (21:9)',
+    ['w'] = 2560,
+    ['h'] = 1080
+  },
+  ['2K'] = {
+    ['label'] = '2K 2048x1152 (16:9)',
+    ['w'] = 2048,
+    ['h'] = 1152
+  },
+  ['4K'] = {
+    ['label'] = '4K 4096x2304 (16:9)',
+    ['w'] = 4096,
+    ['h'] = 2304
+  }
+}
+
+local framerates = {'15', '16', '23.98', '24', '25', '29,97', '30', '48', '50', '59.94', '60'}
+
+local formats = {
+  ['AVI'] = {
+    ['extension'] = 'avi',
+    ['codecs'] = {'mpeg4', 'h263p', 'h264', 'mpeg2video', 'hevc', 'vp9'}
+  },
+  ['Matroska'] = {
+    ['extension'] = 'mkv',
+    ['codecs'] = {'h264', 'h263p', 'mpeg4', 'mpeg2video', 'hevc', 'vp9'}
+  },
+  ['WebM'] = {
+    ['extension'] = 'webm',
+    ['codecs'] = {'vp9', 'h263p', 'h264', 'mpeg4', 'mpeg2video', 'hevc'}
+  },
+  ['MP4'] = {
+    ['extension'] = 'mp4',
+    ['codecs'] = {'h264', 'h263p', 'mpeg4', 'mpeg2video'}
+  },
+  ['QuickTime'] = {
+    ['extension'] = 'mov',
+    ['codecs'] = {'h264', 'h263p', 'mpeg4', 'mpeg2video'}
+  }
+}
+
+res_list = {}
+for i, v in pairs(resolutions) do
+  table.insert(res_list, v['label'])
+end
+
+table.sort(res_list)
+
+format_list = {}
+for i,v in pairs(formats) do
+  table.insert(format_list, i)
+end
+
+table.sort(format_list)
+
+codec_list = formats['AVI']['codecs']
+table.sort(codec_list)
+
+local function extract_resolution(description)
+  for _, v in pairs(resolutions) do
+    if v['label'] == description then
+      return v['w']..'x'..v['h']
+    end
+  end
+  return '100x100'
+end
+
+---- ENSURE ffmpeg is INSTALLED
+
+dt.print_log(MODULE_NAME .. " - Executable Path Preference: "..df.get_executable_path_preference("ffmpeg"))
+local ffmpeg_path = df.check_if_bin_exists("ffmpeg")
+if not ffmpeg_path then
+  dt.print_error("ffmpeg not found")
+  dt.print("ERROR - ffmpeg not found")
+end
+
+local ffmpeg_path_widget = df.executable_path_widget({'ffmpeg'})
+dt.preferences.register("executable_paths", "ffmpeg",
+  "file",
+  MODULE_NAME .. ': ffmpeg location',
+  'Install location of ffmpeg',
+  "ffmpeg",
+  ffmpeg_path_widget
+)
+
```

Comment:
if you use dt.preferences.register then the user has to start darktable, go to settings, change the preference, kill darktable and restart it.

on the other hand, if you do something like

local timelapse_widgets = {}

if not df.check_if_bin_exists("ffmpeg") then
  table.insert(timelapse_widgets, df.executable_path_widget("ffmpeg"))
end

then add your other widgets to timelapse_widgets and just unpack it into the final widget.  That way if you are on linux, ffmpeg is found in the path and nothing happens, otherwise if it's not found the widget is built and placed at the top of the widgets in the exporter.  Once you set the widget the value is available for use so you don't have to restart darktable.

Reply:
nice! done.